src/rpc/dial.rs

```diff
@@ -88,60 +142,27 @@ impl Service<http::Request<BoxBody>> for ViamChannel {
         match self {
             Self::Direct(channel) => Box::pin(channel.call(request)),
             Self::WebRTC(channel) => {
-                let mut status_code = STATUS_CODE_OK;
-                let channel = channel.clone();
+                let mut channel = channel.clone();
                 let fut = async move {
-                    let (parts, body) = request.into_parts();
-
-                    let stream = channel.new_stream();
-                    let stream_id = stream.id;
-                    let metadata = Some(metadata_from_parts(&parts));
-                    let headers = RequestHeaders {
-                        method: parts
-                            .uri
-                            .path_and_query()
-                            .map(PathAndQuery::to_string)
-                            .unwrap_or_default(),
-                        metadata,
-                        timeout: None,
-                    };
-
-                    if let Err(e) = channel.write_headers(&stream, headers).await {
-                        log::error!("error writing headers: {e}");
-                        channel.close_stream_with_recv_error(stream_id, e);
-                        status_code = STATUS_CODE_UNKNOWN;
-                    }
-
-                    let data = hyper::body::to_bytes(body).await.unwrap().to_vec();
-                    if let Err(e) = channel.write_message(false, Some(stream), data).await {
-                        log::error!("error sending message: {e}");
-                        channel.close_stream_with_recv_error(stream_id, e);
-                        status_code = STATUS_CODE_UNKNOWN;
-                    };
-
-                    let body = match channel.resp_body_from_stream(stream_id) {
-                        Ok(body) => body,
-                        Err(e) => {
-                            log::error!("error receiving response from stream: {e}");
-                            channel.close_stream_with_recv_error(stream_id, e);
-                            status_code = STATUS_CODE_UNKNOWN;
-                            Body::empty()
-                        }
-                    };
-
                     let response = http::response::Response::builder()
                         // standardized gRPC headers.
                         .header("content-type", "application/grpc")
                         .version(Version::HTTP_2);
 
-                    let response = if status_code != STATUS_CODE_OK {
-                        response.header("grpc-status", &status_code.to_string())
-                    } else {
-                        response
-                    };
+                    match channel.new_stream() {
+                        Err(e) => {
+                            log::error!("{e}");
```

Comment:
```suggestion
                            log::error!("error getting new stream: {e}");
```

Reply:
I opted for just printing the error here because I think the error message from `new_stream()` is already pretty descriptive, so adding the prefix there just makes it feel wordy. Happy to chat more if you disagree :)

Reply:
No need! Makes sense to me.